front/src/common/Pathfinding/TypeAndPath.tsx

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+import {
+  Path,
+  PostSearchApiArg,
+  SearchResultItemOperationalPoint,
+  osrdEditoastApi,
+} from 'common/api/osrdEditoastApi';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { getInfraID, getRollingStockID } from 'reducers/osrdconf/selectors';
+import { useDebounce } from 'utils/helpers';
+import cx from 'classnames';
+import { GoAlert, GoTriangleRight } from 'react-icons/go';
+import { loadPathFinding } from 'modules/trainschedule/components/ManageTrainSchedule/helpers/adjustConfWithTrainToModify';
+import { setFailure } from 'reducers/main';
+import bbox from '@turf/bbox';
+import { Position } from 'geojson';
+
+type SearchConstraintType = (string | number | string[])[];
+type PathfindingProps = {
+  zoomToFeature: (lngLat: Position, id?: undefined, source?: undefined) => void;
+};
+
+const monospaceOneCharREMWidth = 0.6225;
+
+function OpTooltips({ opList }: { opList: SearchResultItemOperationalPoint[] }) {
+  // Calculation of chars distance from left to put tooltip on center of op name
+  const calcLeftMargin = (charsFromLeft: number, length: number) =>
+    charsFromLeft * monospaceOneCharREMWidth + (length * monospaceOneCharREMWidth) / 2;
+  let charsFromLeft = 0;
+  return (
+    <div className="op-tooltips">
+      {opList.map((op, idx) => {
+        const leftMargin = calcLeftMargin(charsFromLeft, op.trigram.length);
+        charsFromLeft = charsFromLeft + op.trigram.length + 1;
+        return (
+          op.trigram !== '' && (
+            <div
+              className={cx('op', { wrong: !op.name })}
+              key={`typeandpath-op-${idx}-${op.trigram}`}
+              style={{ left: `${leftMargin}rem` }}
+            >
+              {op.name ? op.name : <GoAlert />}
+            </div>
+          )
+        );
+      })}
+    </div>
+  );
+}
+
+export default function TypeAndPath({ zoomToFeature }: PathfindingProps) {
+  const dispatch = useDispatch();
+  const [inputText, setInputText] = useState<string>('');
+  const [opList, setOpList] = useState<SearchResultItemOperationalPoint[]>([]);
+  const infraId = useSelector(getInfraID);
+  const rollingStockId = useSelector(getRollingStockID);
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
```

Comment:
To stick with our convention
```suggestion
  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
```

Reply:
Done !